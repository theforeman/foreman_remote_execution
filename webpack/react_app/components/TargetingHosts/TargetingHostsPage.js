import React from 'react';
import PropTypes from 'prop-types';
import { Grid } from 'patternfly-react';

import SearchBar from 'foremanReact/components/SearchBar';
import Pagination from 'foremanReact/components/Pagination/PaginationWrapper';
import { getControllerSearchProps } from 'foremanReact/constants';

import TargetingHosts from './TargetingHosts';
import './TargetingHostsPage.scss';

const TargetingHostsPage = ({
  handleSearch,
  searchQuery,
  apiStatus,
  items,
  totalHosts,
  pagination,
  handlePagination,
}) => (
  <div id="targeting_hosts">
    <Grid.Row>
      <Grid.Col md={6} className="title_filter">
        <SearchBar
          onSearch={query => handleSearch(query)}
          data={{
            ...getControllerSearchProps('hosts'),
            autocomplete: {
              id: 'targeting_hosts_search',
              searchQuery,
              url: '/hosts/auto_complete_search',
              useKeyShortcuts: true,
            },
            bookmarks: {},
          }}
        />
      </Grid.Col>
    </Grid.Row>
    <br />
    <TargetingHosts apiStatus={apiStatus} items={items} />
    <Pagination
      viewType="table"
      itemCount={totalHosts}
      pagination={pagination}
      onChange={args => handlePagination(args)}
      dropdownButtonId="targeting-hosts-pagination-dropdown"
      className="targeting-hosts-pagination"
    />
  </div>
);

TargetingHostsPage.propTypes = {
  handleSearch: PropTypes.func.isRequired,
  searchQuery: PropTypes.string.isRequired,
  apiStatus: PropTypes.string,
  items: PropTypes.array.isRequired,
  totalHosts: PropTypes.number.isRequired,
  pagination: PropTypes.object.isRequired,
  handlePagination: PropTypes.func.isRequired,
};

TargetingHostsPage.defaultProps = {
  apiStatus: null,
};

export default TargetingHostsPage;
