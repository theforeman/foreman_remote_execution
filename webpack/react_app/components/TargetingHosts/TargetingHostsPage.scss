#targeting_hosts {
  min-height: 350px;
}
