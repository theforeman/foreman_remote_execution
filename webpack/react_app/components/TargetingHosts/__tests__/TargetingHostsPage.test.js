import { testComponentSnapshotsWithFixtures } from '@theforeman/test';
import TargetingHostsPage from '../TargetingHostsPage';
import { TargetingHostsPageFixtures } from './fixtures';

describe('TargetingHostsPage', () =>
  testComponentSnapshotsWithFixtures(
    TargetingHostsPage,
    TargetingHostsPageFixtures
  ));
