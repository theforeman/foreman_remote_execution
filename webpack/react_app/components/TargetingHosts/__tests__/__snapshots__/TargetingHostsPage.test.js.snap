// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TargetingHostsPage renders 1`] = `
<div
  id="targeting_hosts"
>
  <Row
    bsClass="row"
    componentClass="div"
  >
    <Col
      bsClass="col"
      className="title_filter"
      componentClass="div"
      md={6}
    >
      <SearchBar
        data={
          Object {
            "autocomplete": Object {
              "id": "targeting_hosts_search",
              "searchQuery": "",
              "url": "/hosts/auto_complete_search",
              "useKeyShortcuts": true,
            },
            "bookmarks": Object {},
            "controller": "hosts",
          }
        }
        onSearch={[Function]}
      />
    </Col>
  </Row>
  <br />
  <TargetingHosts
    apiStatus="RESOLVED"
    items={
      Array [
        Object {
          "actions": Array [],
          "link": "/link",
          "name": "host",
          "status": "success",
        },
        Object {
          "actions": Array [],
          "link": "/link2",
          "name": "host2",
          "status": "success",
        },
      ]
    }
  />
  <PaginationWrapper
    className="targeting-hosts-pagination"
    dropdownButtonId="targeting-hosts-pagination-dropdown"
    itemCount={1}
    onChange={[Function]}
    pagination={
      Object {
        "page": 1,
        "perPage": 20,
      }
    }
    viewType="table"
  />
</div>
`;
